Library/Homebrew/formula.rb

```diff
@@ -1455,16 +1455,6 @@ def system(cmd, *args)
       $stdout.flush
 
       unless $?.success?
-        log_lines = ENV["HOMEBREW_FAIL_LOG_LINES"]
-        log_lines ||= "15"
-
-        log.flush
-        if !verbose || verbose_using_dots
-          puts "Last #{log_lines} lines from #{logfn}:"
-          Kernel.system "/usr/bin/tail", "-n", log_lines, logfn
-        end
-        log.puts
-
```

Comment:
I think we want to retain this functionality, as it is [used on our bots](https://github.com/Homebrew/homebrew/blob/c77a5848e9918209c516851db05b3fcbd4e7c68c/Library/Homebrew/cmd/test-bot.rb#L836-L842). It is particularly useful on Travis, where it is the only information we have, if a build fails.

My suggestion would be to guard the entire block with

```ruby
if !$?.success? && ARGV.homebrew_developer?
```

instead of the current `unless $?.success?`. This would have the advantage that this functionality is retained for us maintainers, that know how to interpret the information. (Sometimes those lines are enough to see what's wrong.)